src/compiler/factory.ts

```diff
@@ -1544,8 +1544,8 @@ namespace ts {
     export function createIf(expression: Expression, thenStatement: Statement, elseStatement?: Statement) {
         const node = <IfStatement>createSynthesizedNode(SyntaxKind.IfStatement);
         node.expression = expression;
-        node.thenStatement = thenStatement;
-        node.elseStatement = elseStatement;
+        node.thenStatement = isNotEmittedOrPartiallyEmittedNode(thenStatement) ? setTextRange(setOriginalNode(createEmptyStatement(), thenStatement), thenStatement) : thenStatement;

```

Comment:
Shouldn't we do this for any "embedded statement" as well (i.e. `for`, `do`, `while`, labeled statement, etc.)

Reply:
Hm, probably. Should I write up some more test cases and generalize a helper, `asEmbeddedStatement` that is this ternary?

Reply:
Yeah, but you probably just want `isNotEmittedStatement`. A `PartiallyEmittedExpression` is an "expression" and wouldn't be legal in this context (it basically just wraps an expression to add comments/sourcemaps from a parent node that was elided, like eliding parens in a cast `(x as foo)`).